components/service/glean/src/main/java/mozilla/components/service/glean/StringListMetricType.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import kotlinx.coroutines.launch
+import mozilla.components.service.glean.storages.StringListsStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for recording string list metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The string list API exposes the [add] and [set] methods, which take care of validating the input
+ * data and making sure that limits are enforced.
+ */
+data class StringListMetricType(
+    override val disabled: Boolean,
+    override val category: String,
+    override val lifetime: Lifetime,
+    override val name: String,
+    override val sendInPings: List<String>
+) : CommonMetricData {
+
+    override val defaultStorageDestinations: List<String> = listOf("metrics")
+
+    private val logger = Logger("glean/StringListMetricType")
+
+    companion object {
+        // Maximum length of any passed value string, in UTF8 byte sequence length.
+        const val MAX_STRING_LENGTH = 50
+    }
+
+    /**
+     * Appends a string value to one or more string list metric stores.  If the string exceeds the
+     * maximum string length, defined as [MAX_STRING_LENGTH], it will be truncated.
+     *
+     * If adding the string to the lists would exceed the maximum value defined as
+     * [StringListsStorageEngine.MAX_LIST_LENGTH_VALUE], then the storage engine will drop the new
+     * value and it will not be added to the list.
+     *
+     * @param value This is a user defined string value. The maximum length of
+     * this string is [MAX_STRING_LENGTH].
```

Comment:
nit: yours is probably the right way/better way to have multiline comments for `@param`. However, all the glean code uses a slightly different style for it :( Any chance you could change the lines after the first one to indent after the variable name? [see this example](https://github.com/mozilla-mobile/android-components/blob/4b35d14b655c90cfa6ff3c6b675a013a1f868eb3/components/concept/toolbar/src/main/java/mozilla/components/concept/toolbar/Toolbar.kt#L302-L304)

Reply:
This was resolved but I don't see the fix: was it left out of the commit?

Reply:
Haha, nope I must have accidentally Ctrl-Z'd over it or something.  I did it and now it's undone.  Fixing ASAP